oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2Authority.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.core;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.springframework.security.core.AuthenticatedPrincipal;
+import org.springframework.security.core.GrantedAuthority;
+
+public interface OAuth2Authority extends AuthenticatedPrincipal {
```

Comment:
I agree, this name collides with `GrantedAuthority` causing confusion.

I'm concerned with having the word "User" in the name of the principal.

It may be helpful to consider the specs in question. The [JWT RFC](https://tools.ietf.org/html/rfc7519#section-4.1.2) states that the sub claim is:

> The "sub" (subject) claim identifies the principal that is the subject of the JWT.

And the [Introspection RFC](https://tools.ietf.org/html/rfc7662#section-2.2) states that it is based off of JWT:

> Subject of the token, as defined in JWT [RFC7519].

Given that both specs, then, refer to this as the principal, I've changed the name of this class to `OAuth2AuthenticatedPrincipal`.

It may make sense for `OAuth2User` to extend the `OAuth2AuthenticatedPrincipal` interface, if this name turns out to make sense.